components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImpl.java

```diff
@@ -155,12 +155,39 @@ public EndpointRegistryEntry getEndpointRegistryEntryByUUID(String registryId, S
      */
     public String addEndpointRegistryEntry(EndpointRegistryEntry registryEntry) throws APIManagementException {
         if (apiMgtDAO.isRegistryEntryNameExists(registryEntry)) {
-            APIUtil.handleResourceAlreadyExistsException("Endpoint Registry with name '" + registryEntry.getName()
-                    + "' already exists");
+            APIUtil.handleResourceAlreadyExistsException("Endpoint Registry Entry with name '"
+                    + registryEntry.getName() + "' already exists");
         }
         return apiMgtDAO.addEndpointRegistryEntry(registryEntry);
     }
 
+    /**
+     * Updates Registry Entry
+     *
+     * @param registryEntry EndpointRegistryEntry
+     * @throws APIManagementException if failed to update EndpointRegistryEntry
+     */
+    public void updateEndpointRegistryEntry(EndpointRegistryEntry registryEntry) throws APIManagementException {
+        EndpointRegistryEntry endpointRegistryEntry =
+                apiMgtDAO.getEndpointRegistryEntryByUUID(registryEntry.getEntryId());
+        if (!endpointRegistryEntry.getName().equals(registryEntry.getName()) &&
```

Comment:
I think we decided to allow to edit the entry name as well.

Reply:
that is right. this check is only to avoid an existing entry name